packages/edit-post/src/components/preferences-modal/test/index.js

```diff
@@ -20,20 +20,20 @@ jest.mock( '@wordpress/compose/src/hooks/use-viewport-match', () => jest.fn() );
 
 describe( 'EditPostPreferencesModal', () => {
 	describe( 'should match snapshot when the modal is active', () => {
-		it( 'large viewports', () => {
+		it( 'large viewports', async () => {
 			useSelect.mockImplementation( () => [ true, true, false ] );
 			useViewportMatch.mockImplementation( () => true );
 			render( <EditPostPreferencesModal /> );
 			expect(
-				screen.getByRole( 'dialog', { name: 'Preferences' } )
+				await screen.findByRole( 'dialog', { name: 'Preferences' } )
 			).toMatchSnapshot();
```

Comment:
This assertion is still failing for some reason. The snapshot expects the two inactive tabs to have `tabindex="-1", but that attribute isn't present in the test result.

When I test it manually and inspect the elements myself, they have the expected `tabindex`. I'm not sure why the editor preferences modal is rendering properly for me, but not for the test 🤔

Reply:
updated the snapshot to account for ariakit's rendering patterns and added a component-level unit test to ensure tabindex and tab key behavior are correct (ie, pressing 'Tab' doesn't focus the next tab in the tablist)